### PR TITLE
Rename `EditionId` to `RecordId` in the Graph

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

Applies the same changes as #2048 but on the Graph, so the terms do match.

## 🔗 Related links

- Follow-up of https://github.com/hashintel/hash/pull/2048
- Part of [this Asana Task](https://app.asana.com/0/1202805690238892/1203835712237857/f) (_internal_)

## 🔍 What does this change?

- Renames `EntityEditionId` to `EntityRecordId`
    - Renames the fields inside it to be consistent with the 0.3 branch of the BP
- Renames `OntologyTypeEditionId` to `OntologyTypeRecordId`
    - Renames the fields inside it to be consistent
- Introduces correct `VertexId` types
- Adjust mapping functions.

### CODE REVIEW

File: apps/hash-graph/lib/graph/src/store/postgres/query/statement/select.rs [hunk 1]
```diff
@@ -845,40 +845,7 @@ mod tests {
...
-                    &kernel,
-                    &time_projection.image(),
-                    &entity_id.owned_by_id().as_uuid(),
-                    &entity_id.entity_uuid().as_uuid(),
-                ],
-            );
-        }
-
-        #[test]
-        fn for_entity_by_edition_id() {
```
Comment (TimDiekmann) [inline apps/hash-graph/lib/graph/src/store/postgres/query/statement/select.rs:871]: This was an exact copy of the above

File: libs/@local/hash-subgraph/src/temp/map-vertices.ts [hunk 2]
```diff
@@ -83,13 +83,7 @@ const mapEntityType = (entityType: EntityTypeGraphApi): EntityType => {
...
   metadata: OntologyElementMetadataGraphApi,
 ): OntologyElementMetadata => {
-  return {
-    ...metadata,
-    recordId: {
-      baseUri: metadata.editionId.baseId,
-      version: metadata.editionId.version,
-    },
-  };
+  return metadata;
```
Comment (Alfred-Mountfield) [inline libs/@local/hash-subgraph/src/temp/map-vertices.ts:86]: Just flagging for posterity that these are left here purposefully by Tim as directed by me. I intend to remove these in a follow-up PR, but there will be a few more maps we need to do around entity versioning and it'll be good to have this boilerplate in place to put some things in in the interim

**Review by Alfred-Mountfield — COMMENTED** [review]
Thanks Tim, looks good and runs locally. There's one thing that seems to have been missed (I would have expected or hoped for a test failure or error somewhere because of it so I'm a little confused).

Also as a side note, have you made sure that all the params in structural queries are up to date, both in the HTTP tests and backend? (I'd presume so because things seem to be running well, but doesn't hurt to ask)

File: apps/hash-graph/lib/graph/src/store/postgres.rs [hunk 3]
```diff
@@ -25,9 +25,9 @@ pub use self::pool::{AsClient, PostgresStorePool};
 use crate::{
     identifier::{
         account::AccountId,
-        ontology::OntologyTypeEditionId,
+        ontology::OntologyTypeRecordId,
         time::{ProjectedTime, TimeIntervalBound, Timestamp},
-        EntityVertexId,
+        EntityVertexId, OntologyTypeVertexId,
```
[the remark this answers is not in the document]
Comment (TimDiekmann) [reply apps/hash-graph/lib/graph/src/store/postgres.rs:30]: What is strange about this formatting?

[the remark this answers is not in the document]
Comment (Alfred-Mountfield) [reply apps/hash-graph/lib/graph/src/store/postgres.rs:30]: I thought they'd be on newlines, maybe I'm going crazy 😅

**Comment by TimDiekmann** [comment]
> have you made sure that all the params in structural queries are up to date, both in the HTTP tests and backend?

I don't see how structural query parameters are related to these changes.

File: apps/hash-graph/lib/graph/src/knowledge/entity/query.rs [hunk 4]
```diff
@@ -256,7 +256,7 @@ impl fmt::Display for EntityQueryPath<'_> {
             Self::Uuid => fmt.write_str("uuid"),
             Self::OwnedById => fmt.write_str("ownedById"),
             Self::UpdatedById => fmt.write_str("updatedById"),
-            Self::RecordId => fmt.write_str("recordId"),
+            Self::EditionId => fmt.write_str("recordId"),
```
[the remark this answers is not in the document]
Comment (Alfred-Mountfield) [reply apps/hash-graph/lib/graph/src/knowledge/entity/query.rs]: Cheers.

Keeping this conversation open to thread a response to this:
> I don't see how structural query parameters are related to these changes.

Isn't this exactly how params are related to these changes? The entity query paths have changed from `recordId` to `editionId`

[the remark this answers is not in the document]
Comment (TimDiekmann) [reply apps/hash-graph/lib/graph/src/knowledge/entity/query.rs]: You are correct, that the token for `"recorded"` changed to `"editionId"`, but it was never used at any place. I've checked this, and I'm sorry for the confusion.

This comment, however, is on a `Display` implementation and is entirely unrelated to the deserialization of a token.

**Review by Alfred-Mountfield — APPROVED** [review]

File: apps/hash-graph/lib/graph/src/knowledge/entity/query.rs (hunk 4 above)
[the remark this answers is not in the document]
Comment (Alfred-Mountfield) [reply apps/hash-graph/lib/graph/src/knowledge/entity/query.rs]: Thanks for checking

**Pull request merged**